libraries/chain/include/eosio/chain/block_handle.hpp

```diff
@@ -22,6 +24,8 @@ struct block_handle {
    explicit block_handle(block_state_legacy_ptr bsp) : _bsp(std::move(bsp)) {}
    explicit block_handle(block_state_ptr bsp) : _bsp(std::move(bsp)) {}
 
+   bool is_valid() const { return _bsp.index() != std::variant_npos && std::visit([](const auto& bsp) { return !!bsp; }, _bsp); }
```

Comment:
Why not use `valueless_by_exception()` instead of `index()`?

Reply:
`index()` seems clearer in this case to me.

Reply:
So you are saying that there is no point in having `valueless_by_exception()`? I think it was added because it clearly states what we are checking for.